python/tk_multi_publish2/api/item.py

```diff
@@ -0,0 +1,416 @@
+# Copyright (c) 2018 Shotgun Software Inc.
```

Comment:
This existed before here: https://github.com/shotgunsoftware/tk-multi-publish2/blob/master/python/tk_multi_publish2/processing/item.py

I tried to strip it down to eliminate stuff that was UI-specific. The interface has to be the same, but some of the implementations have changed. Still work to do here as you can see at the end of the file.